src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java

```diff
@@ -49,4 +50,12 @@
     @Query("SELECT s FROM Subscription s where s.endDate > CURRENT_TIMESTAMP " +
         "AND s.subscriptionId = :subscriptionId")
     Optional<Subscription> findActiveSubscription(@Param("subscriptionId") String subscriptionId);
+
+    @Query("SELECT s FROM Subscription s WHERE s.accountNumber = :accountNumber AND " +
+        "s.sku = ALL (SELECT DISTINCT o.sku FROM Offering o WHERE " +
+        ":#{#key.usage} = o.usage AND " +
+        ":#{#key.sla} = o.serviceLevel AND " +
+        ":#{T(java.lang.Integer).parseInt(#key.productId)} MEMBER OF o.productIds)")
```

Comment:
@awood FYI I am realizing on doing some smoketesting that the productIds here are different.... key.productId is a swatchProductId, while Offering.productId is an engineering productId... We'll have to reconcile this somehow... will discuss internally.